system/jlib/jsocket.cpp

```diff
@@ -2040,10 +2040,14 @@ size32_t CSocket::writetms(void const* buf, size32_t size, unsigned timeoutms)
     {
         size32_t amnt = write(p,nleft);
 
-        if ( ((amnt == (size32_t)-1) || (amnt == 0)) && (++rollover >= 20) )
+        // can nonblock mode write() return -1 ?
+        if ( (amnt == 0) || (amnt == (size32_t)-1) )
```

Comment:
Checking with @mckellyln re. JIRA comment : 
```
in writemts(), if amnt from write() was ever -1 (currently don't think this is possible) then nwritten, nleft and p should not be incremented (decremented) But still code should be corrected.
```

it appears to me that if amnt == -1 - it would fall into the Sleep block.

Reply:
I misread original code, didn't see enclosing brackets around 2 amnt expressions.